### DISCUSSION TITLE
Problems with sync using onedrive

### DISCUSSION THREAD

**Discussion by JD-aus** [keepassxreboot/keepassxc#6934] [Q&A]
Hi All,

I started using KeepassXC as my first KeePass client, but after having issues with missing entries when using the file directly out of one drive, I found Keepass2 and use the "one drive sync" plugin and have had zero missed entries since then (months now).

My setup is that I use several machines daily and have many others that I need to access or spin up for supporting clients (mainly Win10) at home and office and android (using the fantastic Keepass2Android). I can deal with not having keepassxc on most machines, but need it on my home laptop, home workstation and work workstation.

However, I would prefer to use KeepassXC as I like the idea of having the ability to "share" folders with other users at work (which works OK once set up and you understand that the share files can be deleted at any time and have backups), as well as many other features of XC.

I have read a response somewhere (cannot find it now of course) saying that sync works fine with KeepassXC when you use it correctly. can someone please explain the best way to use cloud sync like one drive etc ... should we be opening the file directly in the one drive folder or is there a better way?

I also have an issue with my "KeePassHttp Settings" being wiped each time I open keepassxc on work machine (to sync the folders with share files for others to use)

Cheers for creating a great open source project that promotes security, we need more open source projects that are this polished and supported.
Jim

**Comment by droidmonkey** [comment]
I use onedrive and keepassdx on android and have never had a sync issue. If you are missing entries and don't see conflict files in your onedrive then that means it wasn't syncing the database. You should open the file directly within the onedrive directory.

**Reply by JD-aus** [reply]
If I look at the versions of the file in onedrive, I can see the files being saved, it is like keepassxc is not seeing the updated file and asking to reload, it is just overriding the new file with the old version it has opened or something.

**Reply by droidmonkey** [reply]
Double check that keepassxc is actually opening the database that is located in onedrive. Also make sure you didn't disable the setting to automatically reload the database when changed externally.

**Reply by JD-aus** [reply]
Hi droidmonkey, thanks for the reply. I haven't been in the office for several weeks, so I was unable to test thoroughly.

I can confirm no settings have been altered, and yes, I have checked to see that the settings checkbox is still checked. I can confirm I am opening the file in onedrive.

I'm happy to enable debug logging if you want to see logs that might help. I have just started using it again after months of using keepass (original) with onedrive plugins

**Reply by Airwolf1544** [reply]
We use KeepassXC in a Team of 4 People.. Database is located on a Sharepoint Site. Yesterday i entered 4 new Passwords.. non of them existet today, so they got lost. It's an major flaw this can happen..

**Reply by JD-aus** [reply]
I actually haven't had any problems in quite a while.

Check the versions of the file to see if you have two versions saved for the day. 

I found that the problem is not the software, but the sync. I also setup backups to be saved each time I saved the dB file and that way I have something to go back to... that option has to be set on each machine unfortunately.

**Reply by adlentis** [reply]
This still isn't solved. Due to the way KeepassXC or OneDrive work, the file actually got deleted here. Fortunately, OneDrive has a recovery feature. I'm going to look for another app or onedrive tool or sync-plugin (like the android app and windows app use).
